From be4f0290b537492d64b0249c32b87dd151f5bc12 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sat, 09 Jul 2022 17:35:22 +0300
Subject: [PATCH] [SW-94367] gaudi3: move setting pcie id to controller config

As recommended by arch team, this patch moves setting of pcie id
in pcie routine as part of PCIe controller configuration section.
The assumption here is made that if PCIe link is already up, ie.
ltssm is set to 1, the pcie device and vendor id are already configured.

Change-Id: I2b13a12deee2c4fd5eae9f7ebde1ba23e38a41e2
---

diff --git a/common/pcie.c b/common/pcie.c
index ec0f752..95e8a83 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -85,8 +85,6 @@
 
 	pcie_loop_while_prstn(PCIE_PRSTN_ASSERT);
 
-	pcie_set_pcie_device_vendor_id();
-
 	ltssm_en = regmap_readl(mmLTSSM_EN) &
 				LTSSM_EN_IND_MASK;
 
@@ -117,6 +115,8 @@
 
 		pcie_phy_set_sram_ext_ld_done();
 
+		pcie_set_pcie_device_vendor_id();
+
 		if ((arcboot_rom_global.config_fuse & EFUSE_SYSCFG_PCIE_GEN_LIMIT) == 0) {
 			arcboot_rom_status_update(PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1);
 			/* limit PCIe link gen to 1 */
